x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/profile/ProfileCancellationIntegTests.java

```diff
@@ -101,7 +102,11 @@ public void onFailure(Exception exception) {
                 taskActions,
                 hasItems(equalTo(SuggestProfilesAction.NAME), equalTo(SearchAction.NAME), startsWith(SearchAction.NAME))
             );
-            tasks.forEach(t -> taskIds.add(t.getId()));
+            assertThat(isShardSearchBlocked(), is(true));
```

Comment:
In addition to make sure tasks are all registered, this new assertion makes sure the shardSearch request is actually handled. This means the test should always wait for the log line "blocking search on ..." before proceeding. If this does not work, at least we can be sure that the problem is in the request handling part instead of cancellation.